core/tests/protractor_desktop/extensions.js

```diff
@@ -128,12 +129,24 @@ describe('Interactions', function() {
 
     for (var interactionId in interactions.INTERACTIONS) {
       var interaction = interactions.INTERACTIONS[interactionId];
+
       for (var i = 0; i < interaction.testSuite.length; i++) {
         var test = interaction.testSuite[i];
 
         explorationEditorMainTab.setInteraction.apply(
           null, [interactionId].concat(test.interactionArguments));
 
+        var deleteResponseButton = element(by.css(
+          '.protractor-test-delete-response'));
+        var confirmDeleteResponseButton = element(by.css(
+          '.protractor-test-confirm-delete-response'));
+        deleteResponseButton.isPresent().then(function(isPresent) {
+          if (isPresent) {
+            deleteResponseButton.click();
+            confirmDeleteResponseButton.click();
+          }
+        });
```

Comment:
We're deleting any previous rules that might have been cached from previous tests of the same interaction so that they won't interfere with the current test. Added a comment clarifying this!